map_model/src/objects/traffic_signals.rs

```diff
@@ -113,7 +138,34 @@ impl ControlTrafficSignal {
                 assert!(m.turn_type != TurnType::Crosswalk);
             }
         }
-
+        Ok(self)
+    }
+    pub fn validate_stage_timings(&self) -> Result<&ControlTrafficSignal, String> {
+        // Is there enough time in each stage to walk across the crosswalk
+        let mut stage_index = 0;
+        for stage in &self.stages {
+            let min_crossing_time = self.get_min_crossing_time(stage_index);
+            if min_crossing_time < stage.phase_type.simple_duration() {
+                // Warn/Error does not seem to be imported?
```

Comment:
Woops, I guess not yet. I'll spend some time this weekend cleaning up logging a bit.

But here, we can just return the `Err`. Depending on the caller (map importer or the UI), maybe it's useful to log the error, or it's just noise.

Reply:
Yeah, I'll leave the println! in for the moment, until you clean up logging